@here/olp-sdk-core/lib/utils/TileKey.ts

```diff
@@ -0,0 +1,417 @@
+ * single tile. On every level, each tile is divided into four children (therefore the name
+ * quadtree).
+ *
+ * Within each [[level]], any particular tile is addressed with [[row]] and [[column]]. The number
+ * of rows and columns in each level is 2 to the power of the level. This means: On level 0, only
+ * one tile exists, [[columnsAtLevel]]() and [[rowsAtLevel]]() are both 1. On level 1, 4 tiles
+ * exist, in 2 rows and 2 columns. On level 2 we have 16 tiles, in 4 rows and 4 columns. And so on.
+ *
+ * A tile key is usually created using [[fromRowColumnLevel]]() method.
+ *
+ * `TileKey` instances are immutable, all members return new instances of `TileKey` and do not
+ * modify the original object.
+ *
+ * Tile keys can be created from and converted into various alternative formats:
+ *
+ *  - [[toQuadKeyString]]() / [[fromQuadKeyString]]() - string representation 4-based
+ *  - [[toHereTile]]() / [[fromHereTile]]() - string representation 10-based
+ *  - [[toMortonCode]]() / [[fromMortonCode]]() - number representation
+ *
+ * Note - as JavaScript's number type can hold 53 bits in its mantissa, only levels up to 26 can be
+ * represented in the number representation returned by [[mortonCode]]().
+ */
+export class TileKey implements QuadKey {
+    private m_mortonCode?: number;
+    private m_hereTile?: string;
+
+    /**
+     * Creates a tile key.
+     *
+     * @param row The requested row. Must be less than 2 to the power of level.
+     * @param column The requested column. Must be less than 2 to the power of level.
+     * @param level The requested level.
+     */
+    static fromRowColumnLevel(
+        row: number,
+        column: number,
+        level: number
+    ): TileKey {
+        return new TileKey(row, column, level);
+    }
+
+    /**
+     * Creates a tile key from a quad string.
+     *
+     * The quad string can be created with [[toQuadKey]].
+     *
+     * @param quadkey The quadkey to convert.
+     * @returns A new instance of `TileKey`.
+     */
+    static fromQuadKeyString(key: string): TileKey {
+        const level = key.length;
+        let row = 0;
+        let column = 0;
+        // tslint:disable:no-bitwise
+        for (let i = 0; i < key.length; ++i) {
+            const mask = 1 << i;
+            const d = parseInt(key.charAt(level - i - 1), 10);
+            if (d & 0x1) {
+                column |= mask;
+            }
+            if (d & 0x2) {
+                row |= mask;
+            }
+        }
+        // tslint:enable:no-bitwise
+        return TileKey.fromRowColumnLevel(row, column, level);
+    }
+
+    /**
+     * Creates a tile key from a numeric Morton code representation.
+     *
+     * You can convert a tile key into a numeric Morton code with [[mortonCode]].
+     *
+     * @param quadKey64 The Morton code to be converted.
+     * @returns A new instance of [[TileKey]].
+     */
+    static fromMortonCode(quadKey64: number): TileKey {
```

Comment:
```suggestion
    static fromMortonCode(key: number): TileKey {
```
Same for all others below. Sounds a little bit more user friendly. `quadKey64` sounds like users need to provide something special. :)

Reply:
updated